commands/aliases.go

```diff
@@ -0,0 +1,203 @@
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/Necroforger/dgwidgets"
+	"github.com/UCCNetworkingSociety/Netsoc-Discord-Bot/logging"
+	"github.com/bwmarrin/discordgo"
+)
+
+var (
+	// note that this is relative to the $(pwd) as opposed to the `commands` directory
+	aliasStorageFilepath = "aliases.json"
+	aliasMap             map[string]string
+)
+
+func init() {
+	aliasMap = make(map[string]string)
+}
+
+// loadFromStorage opens the alias file at `aliasStorageFilepath` and loads
+// the aliases into `aliasMap`.
+func loadFromStorage() error {
+	if _, err := os.Stat(aliasStorageFilepath); os.IsNotExist(err) {
+		if err := ioutil.WriteFile(aliasStorageFilepath, []byte("{}"), 0744); err != nil {
+			return fmt.Errorf("Failed to create alias file: %s", err)
+		}
+	}
+
+	file, err := ioutil.ReadFile(aliasStorageFilepath)
+	if err != nil {
+		return fmt.Errorf("failed to read aliases file: %s", err)
+	}
+
+	if err := json.Unmarshal(file, &aliasMap); err != nil {
+		return fmt.Errorf("failed to unmarshal alias JSON: %s", err)
+	}
+
+	return nil
+}
+
+// writeToStorage overwrites the aliases held within `aliasMap` to the storage file.
+// Note that if an alias was removed from `aliasMap`, the removed alias will also
+// be removed from the storage file.
+func writeToStorage() error {
+	aliasBytes, err := json.Marshal(aliasMap)
+	if err != nil {
+		return fmt.Errorf("Failed to marshal JSON: %s", err)
+	}
+	if err := ioutil.WriteFile(aliasStorageFilepath, aliasBytes, 0744); err != nil {
+		return fmt.Errorf("Failed to write alias file: %s", err)
+	}
+	return nil
+}
+
+// withAliasCommands takes a map of command names to the command itself and
+// adds in a command where the command name is the alias key and the command
+// itself is a command which returns the alias value.
+func withAliasCommands(commMap map[string]Command) (map[string]Command, error) {
+	if err := loadFromStorage(); err != nil {
+		return nil, fmt.Errorf("failed to load aliases from storage: %s", err)
+	}
+
+	for aliasName, aliasValue := range aliasMap {
+		commMap[aliasName] = &textCommand{
+			helpText: aliasValue,
+			command: func(_ context.Context, _ []string) (string, error) {
+				return aliasValue, nil
+			},
+		}
+	}
+	return commMap, nil
+}
+
+// aliasCommand sets string => string shortcut that can be called later to print a value
+func aliasCommand(ctx context.Context, args []string) (string, error) {
```

Comment:
This function could use comments at each `if len(args)` to highlight what the course of action will be

Reply:
Found it a lil hard to follow, not being able to concentrate fully, just a minor thing :P

Reply:
Done :smile: